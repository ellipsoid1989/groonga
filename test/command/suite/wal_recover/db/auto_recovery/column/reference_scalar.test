# _database_close/_database_reopen can't use with HTTP because requests via HTTP are always
# processed by workers. Workers never use primary WAL role.
#@require-interface stdio

#@require-feature message_pack

#$GROONGA_ALLOW_DATABASE_REOPEN=yes
#$GRN_WAL_ROLE=primary

table_create Tags TABLE_HASH_KEY ShortText

table_create Data TABLE_PAT_KEY ShortText
column_create Data tag COLUMN_SCALAR Tags

load --table Data
[
{"_key": "abc",    "tag": "A"},
{"_key": "ABC",    "tag": "A"},
{"_key": "abcdef", "tag": "B"}
]

delete --table Data --key ABC

dump

_database_close

#@copy-path #{db_path}.0000102 #{db_path}.0000102.wal
#@add-important-log-levels notice
_database_reopen
#@remove-important-log-levels notice

dump
