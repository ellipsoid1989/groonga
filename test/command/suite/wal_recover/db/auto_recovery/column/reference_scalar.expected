table_create Tags TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create Data TABLE_PAT_KEY ShortText
[[0,0.0,0.0],true]
column_create Data tag COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
load --table Data
[
{"_key": "abc",    "tag": "A"},
{"_key": "ABC",    "tag": "A"},
{"_key": "abcdef", "tag": "B"}
]
[[0,0.0,0.0],3]
delete --table Data --key ABC
[[0,0.0,0.0],true]
dump
table_create Data TABLE_PAT_KEY ShortText

table_create Tags TABLE_HASH_KEY ShortText

column_create Data tag COLUMN_SCALAR Tags

load --table Data
[
["_key","tag"],
["abc","A"],
["abcdef","B"]
]

load --table Tags
[
["_key"],
["A"],
["B"]
]
_database_close
[[0,0.0,0.0],true]
_database_reopen
[[0,0.0,0.0],true]
#|e| [reader][read-entry][ra][recover][Data.tag] must be map: <positive-integer>(2): path:<db/db.0000102>
#|n| [db][wal][recover] rebuild broken column: <Data.tag>(258)
#|n| DDL:259:column_create Data #recovering#tag
#|n| spec:259:update:Data.#recovering#tag:64(column:fix_size):256(Tags)
#|n| DDL:258:obj_remove Data.tag
#|n| spec:258:remove:Data.tag:64(column:fix_size)
#|n| [db][wal][recover] succeeded to rebuild broken column: <Data.tag>(259)
dump
table_create Data TABLE_PAT_KEY ShortText

table_create Tags TABLE_HASH_KEY ShortText

column_create Data tag COLUMN_SCALAR Tags

load --table Data
[
["_key","tag"],
["abc","A"],
["abcdef","B"]
]

load --table Tags
[
["_key"],
["A"],
["B"]
]
